core/src/main/scala/org/apache/spark/scheduler/OutputCommitCoordinator.scala

```diff
@@ -113,9 +113,11 @@ private[spark] class OutputCommitCoordinator(conf: SparkConf) extends Logging {
         logInfo(
           s"Task was denied committing, stage: $stage, partition: $partition, attempt: $attempt")
       case otherReason =>
-        logDebug(s"Authorized committer $attempt (stage=$stage, partition=$partition) failed;" +
-          s" clearing lock")
-        authorizedCommitters.remove(partition)
+        if (authorizedCommitters.get(partition).exists(_ == attempt)) {
```

Comment:
I think that `authorizedCommitters.exists(_ == (partition, attempt))` might kind of inefficient, too, since I think it's inherited from `Traversable`.

I think this syntax is a bit more confusing, too, so I'm going to leave this as-is and merge it for inclusion in 1.3.1.